components/ui/PortalPopup.tsx

```diff
@@ -0,0 +1,228 @@
+/*
+ * Sonar UI Common
+ * Copyright (C) 2019-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import { throttle } from 'lodash';
+import * as React from 'react';
+import { createPortal, findDOMNode } from 'react-dom';
+import ScreenPositionFixer from '../controls/ScreenPositionFixer';
+import { Popup, PopupPlacement } from './popups';
+
+interface Props {
+  arrowOffset?: number;
+  children: React.ReactNode;
+  className?: string;
+  noPadding?: boolean;
+  overlay: React.ReactNode;
+  placement?: PopupPlacement;
+}
+
+interface Measurements {
+  height: number;
+  left: number;
+  top: number;
+  width: number;
+}
+
+type State = Partial<Measurements>;
+
+function isMeasured(state: State): state is Measurements {
+  return state.height !== undefined;
+}
+
+export default class PortalPopup extends React.Component<Props, State> {
```

Comment:
NP: I wonder if this component shouldn't be added to the `popups.tsx` file, instead of living on its own. It's basically a more advanced variant of the `Popup` component, not its own thing. This feels a bit like the Buttons, which are all in the same file, regardless of their type.